components/servo/lib.rs

```diff
@@ -797,6 +807,13 @@ impl Servo {
             EmbedderMsg::NotifyLoadStatusChanged(webview_id, load_status) => {
                 if let Some(webview) = self.get_webview_handle(webview_id) {
                     webview.set_load_status(load_status);
+                    if let Some((id, ref sender)) =
+                        self.webdriver_senders.borrow().load_status_sender
+                    {
+                        if id == webview_id && load_status == LoadStatus::Complete {
+                            let _ = sender.send(WebDriverLoadStatus::Complete);
+                        }
+                    };
```

Comment:
Instead of integrating this here, please just set a `webdriver_load_status_sender` in `RunningAppState` and respond to load status messages in the `WebViewDelegate` in servoshell.